pixel_dancer.py

```diff
@@ -0,0 +1,693 @@
+
+            # Display warning signs for even number frames
+            elif loop_num % BEAT_CONST == 0:
+                monster.randomize()
+                monster.mode = 2
+                player.energy_color = BLUE
+            else:
+                monster_viewer = MonsterViewer(screen, monster, grid_list)
+            player_viewer = PlayerViewer(screen, player, grid_list)
+            energy_viewer = EnergyViewer(screen, player)
+            rhythm_viewer = RhythmViewer(screen, BEAT_CONST, loop_num,
+                                         MARGINAL_ERROR)
+
+            # update the loop_num(frame number) and let the time flow
+            loop_num += 1
+            clock.tick(FPS)
+            pygame.display.flip()
+
+            # when the enrgy gets to zero, move on to the gameover while loop
+            if player.energy <= 0:
+                running = False
+                gameover = True
+
+        screen = pygame.display.set_mode(canvas_size)
+        while(gameover):
+
+            # displaying messages
+            font = "norasi"
+            font_size = 80
+            msg = "GAME OVER"
+            msg_location = (90, 80)
+
+            font_size2 = 30
+            msg2 = "PRESS ENTER TO PLAY AGAIN"
+            msg_location2 = (90, 480)
+
+            # displays the total number of pictures cleared
+            font_size3 = 50
+            msg3 = "Pictures Cleared : " + str(total_num_pic)
+            msg_location3 = (90, 280)
+            gameover_message = MessageViewer(screen,
+                                             font,
+                                             font_size,
+                                             msg,
+                                             msg_location,
+                                             RED)
+            gameover_message2 = MessageViewer(screen,
+                                              font,
+                                              font_size2,
+                                              msg2,
+                                              msg_location2,
+                                              BLUE)
+            gameover_message3 = MessageViewer(screen,
+                                              font,
+                                              font_size3,
+                                              msg3,
+                                              msg_location3)
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    gameover = False
+
+                # go back to the first while loop to start all over
+                if event.type == pygame.KEYDOWN:
+                    if event.key == pygame.K_RETURN:
+                        gameover = False
+                        start = True
+            clock.tick(FPS)
+            pygame.display.flip()
+    pygame.quit()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
This file is quite long. Maybe think about breaking this into a number of separate modules?